lib/workflows/send_decom_image_email.py

```diff
@@ -0,0 +1,282 @@
+from typing import List, Dict, Optional, Union
+import re
+from datetime import datetime
+from tabulate import tabulate
+from openstack_query import ImageQuery, UserQuery
+
+from enums.cloud_domains import CloudDomains
+from structs.email.smtp_account import SMTPAccount
+from structs.email.email_template_details import EmailTemplateDetails
+from structs.email.email_params import EmailParams
+
+from email_api.emailer import Emailer
+
+
+def get_affected_images_html(image_info_list: List[Dict]):
+    """
+    returns information on images that are to be decommissioned as a html table
+    :param image_info_list: A list of dictionaries containing information on images that are to be decommissioned
+    """
+
+    table_html = "<table>"
+    table_html += "<tr><th>Affected Images</th><th>EOL Date</th><th>Recommended Upgraded Image</th></tr>"
+    for image in image_info_list:
+        table_html += f"<tr><td>{image['name']}</td><td>{image['eol']}</td><td>{image['upgrade']}</td></tr>"
+    table_html += "</table>"
+    return table_html
+
+
+def get_affected_images_plaintext(image_info_list: List[Dict]):
+    """
+    returns information on images that are to be decommissioned as a plaintext tabulate table
+    :param image_info_list: A list of dictionaries containing information on images that are to be decommissioned
+    """
+    return tabulate(
+        image_info_list,
+        headers={
+            "name": "Affected Images",
+            "eol": "EOL Date",
+            "upgrade": "Recommended Upgrade Image",
+        },
+        tablefmt="plain",
+    )
+
+
+def get_image_info(
+    image_name_list: List[str],
+    image_eol_list: List[str],
+    image_upgrade_list: List[str],
+):
+    """
+    function validates key-word arguments related to displaying information on images to decommission and parses it
+    into a list of dictionaries
+    """
+    assert (
+        len(image_name_list) > 0
+    ), "please provide a list of image names to decommission"
+
+    assert len(image_name_list) == len(
+        image_eol_list
+    ), "Lists un-equal: Each image to decommission requires an EOL date"
+
+    assert len(image_name_list) == len(image_upgrade_list), (
+        "Lists un-equal: Each image to decommission requires a recommended upgraded image to replace it, "
+        "leave blank '' to specify if no upgrade image exists (and deletion is recommended)"
+    )
```

Comment:
These should be `if` statements, asserts are for programming errors which shouldn't happen

Instead this should be either a `ValueError` or something similar